src/main/java/seedu/address/logic/commands/NoteCommand.java

```diff
@@ -45,14 +47,34 @@ public NoteCommand(Name name, Note note) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        Person personToEdit = model.findByName(name);
-        if (personToEdit instanceof Maintainer || personToEdit instanceof Staff
-                || personToEdit instanceof Supplier) {
-            throw new CommandException("Person is not of general person class");
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Person personToEdit = findByName(lastShownList, name);
+        Person editedPerson;
+
+        if (personToEdit instanceof Maintainer) {
```

Comment:
same comment as yilin here, you may want to abstract out `NoteCommand` to each individual user class to improve polymorphism